src/server/server_family.cc

```diff
@@ -405,15 +440,48 @@ void ServerFamily::Load(const std::string& load_path) {
 #endif
 
   auto& pool = service_.proactor_pool();
-  // Choose thread that does not handle shards if possible.
-  // This will balance out the CPU during the load.
-  ProactorBase* proactor =
-      shard_count() < pool.size() ? pool.at(shard_count()) : pool.GetNextProactor();
-
-  load_fiber_ = proactor->LaunchFiber([load_path, this] {
-    auto ec = LoadRdb(load_path);
-    LOG_IF(ERROR, ec) << "Error loading file " << ec.message();
-  });
+
+  // Channel to receive error codes from loading fibers.
+  // buffer size is no less than there are load fibers to prevent any blocking.
+  auto chan = std::make_shared<::util::fibers_ext::SimpleChannel<std::error_code>>(paths.size());
```

Comment:
> if you would explicitly keep fibers handles in an array (instead of detaching) you could join on them inside load_join_fiber and get the same outcome

I did this first. But then I realized I need each fibers return value (the error code).

> instead of just using fibers::mutex
I've seen this in your code. I even fixed a bug with the same pattern (but opstatus instead of ec) 😄 

I not really fond of it. First, its error prone (see my comment above). Second, its kind of very simple and limited. No info what else happened, where it originated. In case you wanted to add more debug messages or report the corrupted file, you'd have to do that further down in parsing code. In case you want to detach the fiber after the first error - you can't either.

But I have no problem with having your suggested approach here.

> an opportunity to introduce AggregateError thread-safe class that merges into itself std::error_code if it's set
That seems like a helpful utility. At least you don't have to setup the mutex and overwrite-with-0 checks yourself. 

By the way, the same applies to OpStatus. What is more, its just an enum - it can be edited atomically without any mutex at all

Reply:
Sounds good